### PR TITLE
Fix type annotation for pandas.core.generic.py

### LINKED ISSUE #25901: Fix Type Annotations in pandas.core.generic
Part of #25882

Current errors are as follows:

```sh
pandas/core/generic.py:116: error: Need type annotation for '_accessors'
pandas/core/generic.py:119: error: Need type annotation for '_metadata'
```

Both of these are containers which should contain strings, so just need proper annotation and this module can be removed from mypy.ini

### PR DESCRIPTION
**Pull request by vaibhavhrt**
Also remove this module from mypy.ini

- [x] closes #25901 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -113,10 +114,10 @@ class NDFrame(PandasObject, SelectionMixin):
                        '_default_fill_value', '_metadata', '__array_struct__',
                        '__array_interface__']
     _internal_names_set = set(_internal_names)
-    _accessors = frozenset()
+    _accessors = frozenset()  # type: Container[str]
```
Comment (WillAyd) [inline pandas/core/generic.py]: Should use FrozenSet here

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by WillAyd** [comment]
@vaibhavhrt could you also take care of the following?

```sh
pandas/core/frame.py:368: error: Need type annotation for '_accessors'
```

Very similar problem mentioned in issue I think makes sense to knock out together

File: pandas/core/generic.py (hunk 1 above)
Comment (vaibhavhrt) [reply pandas/core/generic.py]: Sure I will fix that.

**Comment by vaibhavhrt** [comment]
@WillAyd Sure I will take care of frames.py too

**Comment by vaibhavhrt** [comment]
There is a little problem with frame.py, I am getting this error instead of what you wrote in your comment:

> pandas\core\frame.py:368: error: Incompatible types in assignment (expression has type "Set[\<nothing\>]", base class "NDFrame" defined the type as "FrozenSet[str]")

And I can't fix it even after adding type. Still getting the same(almost) error.

> pandas\core\frame.py:368: error: Incompatible types in assignment (expression has type "Set[str]", base class "NDFrame" defined the type as "FrozenSet[str]")

**Comment by WillAyd** [comment]
I think just using Set consistently should be fine.

**Comment by vaibhavhrt** [comment]
Alright I will replace `_accessor = frozenset()` with `_accessor = set()` in generic.NDFrame too.

**Comment by WillAyd** [comment]
@vaibhavhrt yes change both

**Comment by vaibhavhrt** [comment]
I have already changed it to `_accessor = set()  # type: Set[str]` in both files, but it's on my other laptop. I will push them tomorrow morning.

**Review by WillAyd — APPROVED** [review]
lgtm @jreback

File: pandas/core/generic.py [hunk 2]
```diff
@@ -113,10 +114,10 @@ class NDFrame(PandasObject, SelectionMixin):
                        '_default_fill_value', '_metadata', '__array_struct__',
                        '__array_interface__']
     _internal_names_set = set(_internal_names)
```
Comment (jreback) [inline pandas/core/generic.py]: can you type the rest of these?

File: pandas/core/frame.py [hunk 3]
```diff
@@ -365,7 +365,7 @@ def _constructor(self):
     _constructor_sliced = Series
```
Comment (jreback) [inline pandas/core/frame.py]: same here

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 2 above)
Comment (vaibhavhrt) [reply pandas/core/generic.py]: @jreback Sure, I will get it done tomorrow morning(gotta sleep now :sleeping: ).

**Comment by jreback** [comment]
also pls merge master

**Comment by vaibhavhrt** [comment]
@jreback Yeah I saw the conflicts, looks like master updated after I started working on it. I will take care of that too.

File: pandas/core/generic.py (hunk 2 above)
Comment (vaibhavhrt) [reply pandas/core/generic.py]: @jreback done.

File: pandas/core/frame.py (hunk 3 above)
Comment (vaibhavhrt) [reply pandas/core/frame.py]: its done too

**Comment by vaibhavhrt** [comment]
@jreback I have made changes as requested. Please review it when you find some time.

File: pandas/core/frame.py [hunk 4]
```diff
@@ -363,10 +363,11 @@ class DataFrame(NDFrame):
...
 
-    _constructor_sliced = Series
-    _deprecations = NDFrame._deprecations | frozenset(
-        ['get_value', 'set_value', 'from_csv', 'from_items'])
-    _accessors = set()
+    _constructor_sliced = Series  # type: Type[Series]
+    _deprecations = NDFrame._deprecations | frozenset([
+        'get_value', 'set_value', 'from_csv', 'from_items'
+    ])  # type: FrozenSet[str]
+    _accessors = set()  # type: Set[str]
```
Comment (jbrockmendel) [inline pandas/core/frame.py]: @WillAyd Im still not clear on what the upside from this added verbosity is.  Can mypy really not figure out that `set()` is a set?

**Comment by WillAyd** [comment]
It is required because it can’t infer from the declaration what type of objects the set should hold.

**Comment by WillAyd** [comment]
@vaibhavhrt can you merge master?

**Comment by vaibhavhrt** [comment]
@WillAyd master merged.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @vaibhavhrt

**Comment by vaibhavhrt** [comment]
Thanks for merging, I will pick up some new issues tomorrow.

**Pull request merged**